public/app/plugins/datasource/grafana-azure-monitor-datasource/datasource.test.ts

```diff
@@ -34,4 +34,17 @@ describe('Azure Monitor Datasource', () => {
       );
     });
   });
+
+  it('should not filter a valid query', () => {
+    const ds = new Datasource(createMockInstanceSetttings());
+    const query = createMockQuery();
+    expect(ds.filterQuery(query)).toBe(true);
+  });
+
+  it('should not filter query with no query type', () => {
```

Comment:
I think this should be `should filter query with no query type`?

Reply:
It also confuses me that `filterQuery` returns true for a query that *shouldn't* be filtered. That feels counter- intuitive to me?

Reply:
the syntax is complex :sweat_smile: but it's the default behavior in JS, executions that return `true` are kept:

```
Welcome to Node.js v16.17.0.
Type ".help" for more information.
> [true, false].filter(e => e)
[ true ]
> 
```

Reply:
It's always baffled me that this was the default behaviour 😂